code/modules/mob/living/simple_animal/hostile/megafauna/chaos_marine.dm

```diff
@@ -0,0 +1,510 @@
+		blood_dash(target)
+	if(get_dist(src, target) > round(5*dash_mod) && teleport_cooldown <= world.time && !charging)
+		teleport_b(target)
+	if(get_dist(src, target) > round(4) && ranged_cooldown <= world.time && !charging)
+		if(health <= maxHealth*0.5)
+			rapid_fire()
+		else
+			blast()
+	if(get_dist(src, target) <= 3 && runic_blast_cooldown <= world.time && !charging)
+		runic_blast()
+	if(infernal_summon_cooldown <= world.time && !charging)
+		infernal_summon()
+
+/obj/projectile/bullet/chaos_bomb
+	name ="bolter bullet"
+	desc = "Oh god oh fu..."
+	icon_state = "bolter"
+	damage = 20
+
+/obj/projectile/bullet/chaos_bomb/on_hit(atom/target, blocked = FALSE)
+	..()
+	explosion(target, -1, 0, 0, 0, 0, flame_range = 2)
+	return BULLET_ACT_HIT
+
+/obj/projectile/bullet/chaos_bomb/blood
+	name = "blood bolt"
+	icon_state = "mini_leaper"
+	damage = 10
+	nondirectional_sprite = TRUE
+	impact_effect_type = /obj/effect/temp_visual/dir_setting/bloodsplatter
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/ex_act(severity, target)
+	return //Resistant to explosions
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/death(gibbed)
+	if(health > 0)
+		return
+	var/obj/effect/portal/permanent/one_way/exit = new /obj/effect/portal/permanent/one_way(starting)
+	exit.id = "chaos_marine_exit"
+	exit.add_atom_colour(COLOR_RED_LIGHT, ADMIN_COLOUR_PRIORITY)
+	exit.set_light(15, 1, LIGHT_COLOR_RED)
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/proc/blood_dash(target)
+	if(charging || dash_cooldown > world.time)
+		return
+	dash_cooldown = world.time + (initial(dash_cooldown) * dash_mod)
+	charging = TRUE
+	var/dir_to_target = get_dir(get_turf(src), get_turf(target))
+	var/turf/T = get_step(get_turf(src), dir_to_target)
+	for(var/i in 1 to dash_num)
+		new /obj/effect/temp_visual/dragon_swoop/legionnaire(T)
+		T = get_step(T, dir_to_target)
+	addtimer(CALLBACK(src, .proc/blood_dash_2, dir_to_target, 0), (5 * dash_mod))
+	playsound(src,'sound/effects/meteorimpact.ogg', 200, 1)
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/proc/blood_dash_2(var/move_dir, var/times_ran)
+	if(times_ran >= dash_num)
+		charging = FALSE
+		return
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(ismineralturf(T))
+		var/turf/closed/mineral/M = T
+		M.gets_drilled()
+	if(T.density)
+		charging = FALSE
+		return
+	for(var/obj/structure/window/W in T.contents)
+		charging = FALSE
+		return
+	for(var/obj/machinery/door/D in T.contents)
+		if(D.density)
+			charging = FALSE
+			return
+	forceMove(T)
+	playsound(src,"sound/effects/footstep/heavy[pick(1,2)].ogg", 200, 1)
+	for(var/mob/living/L in T.contents - src)
+		if(!faction_check_mob(L))
+			visible_message("<span class='boldwarning'>[src] runs through [L]!</span>")
+			to_chat(L, "<span class='userdanger'>[src] pierces you with a chain-sword!</span>")
+			explosion(L, -1, 0, 2, 0, 0, flame_range = 0)
```

Comment:
a heavy explosion will deal 120 damage to someone without bomb armor, just so you know

and even against someone with bomb armor, it'll knock them out for 2 seconds and knock them down for a considerable amount of time

Reply:
light explosions will also knock you down, making it much harder to dodge future attacks, even if you have an immunity to damage slowdown